OpenRA.Game/World.cs

```diff
@@ -180,7 +180,19 @@ internal World(ModData modData, Map map, OrderManager orderManager, WorldType ty
 			OrderManager = orderManager;
 			orderGenerator = new UnitOrderGenerator();
 			Map = map;
-			Timestep = orderManager.LobbyInfo.GlobalSettings.Timestep;
+
+			// HACK: GlobalSettings.Timestep isn't set at the right time to affect shellmaps,
+			// so we try to get the default gamespeed and apply it here directly.
+			// Necessary to prevent shellmaps from always running at Timestep 40, ignoring all mod-defined gamespeeds.
+			if (type == WorldType.Shellmap)
+			{
+				var defaultGameSpeed = modData.Manifest.Get<GameSpeeds>().Speeds.FirstOrDefault(s => s.Key == "default");
+				if (defaultGameSpeed.Value != null)
+					Timestep = defaultGameSpeed.Value.Timestep;
```

Comment:
It looks like we should be able to do this more cleanly at the bottom of `MapOptions.Created`. This already knows what the mod-defined default speed is, so no need to hardcode "default".

It would also be good to elaborate a bit more about "at the right time", so I suggest the following:
```csharp
// The world timestep and order latency for regular games is hacked as a
// side-effect on top of the regular lobby options plumbing.
// The Shellmap and map editor bypass that plumbing, so we must manually fix
// the game speed here. The order latency is not relevant for these so is ignored.
if (self.World.Type != WorldType.Regular)
	self.World.Timestep = GameSpeed.Timestep;
```

Am i missing something that breaks if we allow the timestep to stay at the incorrect GlobalSettings default between world creation and `MapOptions.Created`?